### DISCUSSION TITLE
Make subset.eq the default variant of subset

### DISCUSSION THREAD

**Discussion by yaksher** [typst/typst#2443] [Ideas]
The title says it all. While Typst's current form follows what LaTeX does, I have never encountered a circumstance where $\subset$ is the variant I want, here or in LaTeX. This is in substantial part because the definition $\subset$ is not standard and is sometimes used to mean any subset, i.e., $\subseteq$ and other times to mean proper subset, i.e., $\subsetneq$. The very few documents I ever see using $\subset$ are ones like lecture notes or whatever where the author was probably too lazy to type $\subseteq$ over $\subset$.

Tl;dr: Due to ambiguity of then meaning of $\subset$ between $\subseteq$ and $\subsetneq$, it's rarely the right choice, and so instead of following LaTeX, `subset` should default to `subset.eq` and a variant should be added for the nonexistent case when somebody means $\subset$.

**Comment by PgBiel** [comment]
@yaksher There has been extensive discussion before regarding names of math symbols. I don't think any change is needed here, as this is just a default, and defaults can't satisfy everybody. The defaults don't satisfy you, but they've satisfied me before, for instance.

Instead, what you're looking for is typst's built-in mechanism to _override_ the defaults, which consists of let bindings. Here's what you can do (for instance) for your particular case:

```typ
#let subset = symbol(
  "⊆",
  ("simple", "⊂")
)

#subset
#subset.simple
```

The above will render:

![image](https://github.com/typst/typst/assets/9021226/aed9314e-e92f-4e46-ad60-93a31792761b)

For reference, see also this section of the docs: https://typst.app/docs/reference/symbols/

**Reply by yaksher** [reply]
Would this also still allow you to do `subset.neq` or does that need to manually be added as well?

**Reply by PgBiel** [reply]
You are overriding the whole symbol, so you'd have to add `neq` and other variants manually, yes.

**Comment by Enivex** [comment]
Speaking as a mathematician, I encounter ⊂ significantly more often than ⊆

**Reply by yaksher** [reply]
Fascinating, I encounter it rarely and it's almost always ambiguous, so I assume people just do it out of being too lazy to type what they actually mean. I believe it used to mean proper subset more consistently, so I guess older papers probably use it more to mean that.

**Reply by Enivex** [reply]
Proper subsets don't appear that often relative to just any old subset, and when they do ⊊ is significantly less ambiguous

**Reply by yaksher** [reply]
Yes, that is exactly my point. $\subset$ is ambiguous and $\subseteq$ works better for improper subset, because it's clear and $\subset$ used to/sometimes means proper subset. And $\subsetneq$ works better for proper subset because half the time people just mean $\subset$.

Reasonable to not bother with changing defaults, but my point is exactly that it's almost always better to use $\subseteq$ or $\subsetneq$, so making the default something that is relatively rarely the correct choice seems odd to me.

**Reply by Enivex** [reply]
> Yes, that is exactly my point

Not really. It's no more ambiguous, because if a text contains ⊊ then it's safe to assume that ⊂ just means subset, not proper subset.

The symbol ⊂ is widely used (arguably more widely used than ⊆) for subset, so trying to make an opinionated change to ⊆ is just going to cause frustration.

Note also that the official unicode name of ⊂ is "Subset Of" https://www.compart.com/en/unicode/U+2282
